### PR TITLE
Add backend for list of editables

### LINKED ISSUE #4416: List of editables: List backend
The most important functionality here is to have an endpoint returning a list of editables with basic information. Feel free to return dummy data for things that are more expensive to get (like the current state / bullet color that require diving into the revisions of the editable).

Also, for performance reasons we need simple pagination, i.e. being able to specify:

- the max number of editables to load
- what to sort by (e.g. ID, title - not sure if anything else makes sense); ID should always be used as a last criterion to make the sorting stable in case of duplicate titles
- before/after id - needed when loading a later/earlier page; if one (both = invalid) is present, you'd only load items with an ID less/greater than what's specified. we may not need "before" at all unless we ditch virtualized rendering in favor of explizit prev/next navigation, or we don't always keep already-loaded data.

### PR DESCRIPTION
**Pull request by ParthS007**
closes #4416 
Screenshot from 2020-06-09 16-19-30

### CODE REVIEW

File: indico/modules/events/editing/controllers/backend/editable_list.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+from indico.modules.events.contributions.models.contributions import Contribution
+from indico.modules.events.editing.controllers.base import RHEditingBase
+from indico.modules.events.editing.models.editable import EditableType
+
+
+class RHEditableList(RHEditingBase):
+    """Return editable list of the event for the editable type."""
+    def _process_args(self):
+        RHEditingBase._process_args(self)
+        self.event = Event.get(int(request.view_args['confId']))
```
Comment (mic4ael) [inline indico/modules/events/editing/controllers/backend/editable_list.py:22]: Not needed since one of the base classes already does that

File: indico/modules/events/editing/controllers/backend/editable_list.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+        self.editable_type = EditableType[request.view_args['type']]
```
Comment (mic4ael) [inline indico/modules/events/editing/controllers/backend/editable_list.py:23]: I guess you can inherit from `RHEditableTypeManagementBase` to have `self.editable_type` as well

File: indico/modules/events/editing/controllers/backend/editable_list.py [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+        self.contributions = Contribution.query.with_parent(self.event)
+        self.is_code_present = False
+        self.serialized_contributions = []
+        for contribution in self.contributions:
+            serialized_editable = None
+            if contribution.editables:
```
Comment (mic4ael) [inline indico/modules/events/editing/controllers/backend/editable_list.py:29]: no need for that `if` since if `contribution.editables` is empty it will not do anything anyway ;) (and we will save some indentation)

File: indico/modules/events/editing/client/js/management/editable_type/EditableList.jsx [hunk 4]
```diff
@@ -153,13 +189,21 @@ export default function EditableList() {
                     />
                   )}
                 />
-                {columnHeaders.map(([key, label, width, extraProps]) => (
+                {code && (
+                  <Column
+                    key="code"
+                    dataKey="code"
+                    label="Code"
```
Comment (mic4ael) [inline indico/modules/events/editing/client/js/management/editable_type/EditableList.jsx:196]: missing i18n

File: indico/modules/events/editing/controllers/backend/editable_list.py [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+                for editable in contribution.editables:
+                    if editable.type.name == self.editable_type.name:
+                        serialized_editable = self.serialize_editable(editable)
+            self.serialized_contributions.append(self.serialize_contribution(contribution, serialized_editable))
```
Comment (mic4ael) [inline indico/modules/events/editing/controllers/backend/editable_list.py:33]: I am not sure if that is the best place to serialize the contributions. Usually we "deserialize" the request arguments to a Pythonic form in `_process_args` which then can be serialized inside RH (what you are doing with `self.contributions`). Maybe to keep the `process_args` small and have only one responsibility you should move the serialization to a separate method which then would be called from `_process`.

Comment (ThiefMaster) [reply indico/modules/events/editing/controllers/backend/editable_list.py:33]: indeed, this is way too much logic for `_process_args` - this all belongs in `_process`. also, i think we could make use of a marshmallow schema

Comment (ParthS007) [reply indico/modules/events/editing/controllers/backend/editable_list.py:33]: Do we need schema here?

Comment (ThiefMaster) [reply indico/modules/events/editing/controllers/backend/editable_list.py:33]: I think using one would certainly be cleaner.

File: indico/modules/events/editing/client/js/management/editable_type/EditableList.jsx [hunk 6]
```diff
@@ -153,13 +189,21 @@ export default function EditableList() {
                     />
                   )}
                 />
-                {columnHeaders.map(([key, label, width, extraProps]) => (
+                {code && (
```
Comment (panagiotappl) [inline indico/modules/events/editing/client/js/management/editable_type/EditableList.jsx:192]: Maybe you could add this as a `columnHeader` conditionally somehow?
I couldn't find any pretty solution though...
Here's an example:
``` jsx
const columnHeaders = [
    ['id', Translate.string('ID'), 60],
    ...code ? [['code', Translate.string('Code'), 80]] : [],
    ['title', Translate.string('Title'), 600],
    ['status', Translate.string('Status'), 200],
    ['editor', Translate.string('Editor'), 400],
  ];
```

File: indico/modules/events/editing/controllers/backend/editable_list.py [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+from flask import jsonify, request
+
+from indico.modules.events import Event
+from indico.modules.events.contributions.models.contributions import Contribution
+from indico.modules.events.editing.controllers.base import RHEditingBase
+from indico.modules.events.editing.models.editable import EditableType
+
+
+class RHEditableList(RHEditingBase):
+    """Return editable list of the event for the editable type."""
```
Comment (panagiotappl) [inline indico/modules/events/editing/controllers/backend/editable_list.py:19]: Maybe you could rephrase this to "Return the list of editables of the event for a given type"

File: indico/modules/events/editing/client/js/management/editable_type/EditableList.jsx (hunk 6 above)
Comment (ParthS007) [reply indico/modules/events/editing/client/js/management/editable_type/EditableList.jsx:192]: It was looking a bit ugly but I can go ahead with this if everyone is fine with this. :+1:

File: indico/modules/events/editing/client/js/editing/timeline/StateIndicator.jsx [hunk 8]
```diff
@@ -20,6 +20,8 @@ const colors = {
   accepted: 'green',
   assigned: 'purple',
   needs_submitter_changes: 'orange',
+  not_submitted: 'blue',
+  ready_for_review: 'olive',
```
[the remark this answers is not in the document]
Comment (ParthS007) [reply indico/modules/events/editing/client/js/editing/timeline/StateIndicator.jsx:24]: What about brown or grey?

File: indico/modules/events/editing/client/js/management/editable_type/EditableList.jsx [hunk 9]
```diff
@@ -66,16 +89,29 @@ export default function EditableList() {
...
-        <div index={rowIndex} styleName="rowcolumn-tooltip" role="gridcell">
-          {dataKey === 'assigned' && <Icon name="user outline" />}
-          {dataKey === 'status' ? (
-            <StateIndicator index={rowIndex} state={sortedList[rowIndex][dataKey]} circular label />
-          ) : (
-            sortedList[rowIndex][dataKey]
-          )}
-        </div>
-      </TooltipIfTruncated>
+      sortedList[rowIndex][dataKey] !== null && (
```
[the remark this answers is not in the document]
Comment (ParthS007) [reply indico/modules/events/editing/client/js/management/editable_type/EditableList.jsx:92]: I checked with zero code, it's working fine. Going ahead with this. :+1:

File: indico/modules/events/editing/schemas.py [hunk 10]
```diff
@@ -139,6 +140,29 @@ class Meta:
     review_conditions_valid = fields.Boolean()
 
 
+class EditableBasicSchema(mm.ModelSchema):
+    class Meta:
+        model = Editable
+        fields = ('id', 'type', 'state', 'editor')
+
+    state = fields.String()
+    editor = fields.Nested(UserSchema, only=('id', 'avatar_bg_color', 'full_name'))
```
Comment (ParthS007) [inline indico/modules/events/editing/schemas.py:149]: Currently, Assigning of editor is not implemented so it will give error.

File: indico/modules/events/editing/schemas.py [hunk 11]
```diff
@@ -139,6 +140,29 @@ class Meta:
...
+class EditingEditableListSchema(mm.ModelSchema):
+    class Meta:
+        model = Contribution
+        rh_context = ('editable_type')
+        fields = ('id', 'friendly_id', 'title', 'code', 'editable')
+
+    editable = fields.Method('_get_editable')
+
+    def _get_editable(self, contribution):
+        editable_type = self.context['editable_type']
```
Comment (ParthS007) [inline indico/modules/events/editing/schemas.py:161]: Why `editable_type` is not available in `self.context`?

Comment (ThiefMaster) [reply indico/modules/events/editing/schemas.py:161]: Look carefully at this line again :)

https://github.com/indico/indico/pull/4460/files#diff-14a5c7d2bef671ab20ad3825f8c7d38aR155

Comment (ParthS007) [reply indico/modules/events/editing/schemas.py:161]: No, that is a different error. Fixed that!

Comment (ThiefMaster) [reply indico/modules/events/editing/schemas.py:161]: The `rh_context` not being a tuple would explain why you don't have the `editable_type` in the context though...

File: indico/modules/events/editing/schemas.py [hunk 12]
```diff
@@ -139,6 +140,29 @@ class Meta:
...
+        editable = {editable for editable in contribution.editables if editable.type.name == editable_type.name}
```
Comment (ThiefMaster) [inline indico/modules/events/editing/schemas.py:162]: why are you comparing by name here?

Comment (ParthS007) [reply indico/modules/events/editing/schemas.py:162]: Comparing whether it is a `paper`, `slides` or `poster`.

File: indico/modules/events/editing/schemas.py (hunk 10 above)
Comment (ThiefMaster) [reply indico/modules/events/editing/schemas.py:149]: Why does it give an error? If there is no editor assigned it should serialize as `None`

File: indico/modules/events/editing/schemas.py (hunk 11 above)
Comment (ParthS007) [reply indico/modules/events/editing/schemas.py:161]: If I did `rh_context = ('editable_type',)`, still gives error.

File: indico/modules/events/editing/schemas.py (hunk 12 above)
Comment (ThiefMaster) [reply indico/modules/events/editing/schemas.py:162]: What I mean is, why not just compare `editable.type == editable_type`

File: indico/modules/events/editing/schemas.py (hunk 11 above)
Comment (ThiefMaster) [reply indico/modules/events/editing/schemas.py:161]: time to add some print statements directly in `use_rh_args` to debug what's going wrong :)

File: indico/modules/events/editing/schemas.py (hunk 10 above)
Comment (ParthS007) [reply indico/modules/events/editing/schemas.py:149]: Resolved!

File: indico/modules/events/editing/client/js/management/editable_type/EditableList.jsx [hunk 13]
```diff
@@ -20,36 +21,67 @@ import {
...
+  });
+
+  let editableList = data || lastData;
+  if (isLoadingEditableList && !lastData) {
+    return <Loader inline="centered" active />;
+  } else if (!editableList) {
+    return null;
+  }
+
+  editableList = [...data];
```
Comment (mic4ael) [inline indico/modules/events/editing/client/js/management/editable_type/EditableList.jsx:46]: What is the point of this?

File: indico/modules/events/editing/controllers/backend/editable_list.py [hunk 14]
```diff
@@ -0,0 +1,28 @@
...
+    """Return the list of editables of the event for a given type"""
+    def _process_args(self):
+        RHEditableTypeManagementBase._process_args(self)
+        self.contributions = (Contribution.query
+                              .with_parent(self.event)
+                              .options(joinedload('editables'))
+                              .all())
+
+    def _process(self):
+        return (EditingEditableListSchema(many=True, context={"editable_type": self.editable_type})
```
Comment (mic4ael) [inline indico/modules/events/editing/controllers/backend/editable_list.py:27]:
```suggestion
        return (EditingEditableListSchema(many=True, context={'editable_type': self.editable_type})
```

File: indico/modules/events/editing/client/js/management/editable_type/EditableList.jsx [hunk 15]
```diff
@@ -20,36 +21,67 @@ import {
...
+  editableList.forEach(contribution => {
+    if (_.isEmpty(contribution.editable)) {
+      contribution.status = 'not_submitted';
+      contribution.editor = null;
+    } else {
+      contribution.status = contribution.editable.state;
+      contribution.editor = contribution.editable.editor.name;
+    }
+    if (!contribution.code) {
+      contribution.code = 'N/A';
```
Comment (mic4ael) [inline indico/modules/events/editing/client/js/management/editable_type/EditableList.jsx:56]: Wouldn't it be better to actually move this to `renderCell` instead of modifying the object directly?

```javascript
{dataKey === 'code' && !code && Translate.string('N/A')}
```

File: indico/modules/events/editing/schemas.py [hunk 16]
```diff
@@ -139,6 +140,28 @@ class Meta:
...
+    class Meta:
+        model = Contribution
+        fields = ('id', 'friendly_id', 'title', 'code', 'editable')
+
+    editable = fields.Method('_get_editable')
+
+    def _get_editable(self, contribution):
+        editable_type = self.context['editable_type']
+        editable = {editable for editable in contribution.editables if editable.type == editable_type}
+        return EditableBasicSchema().dump(editable)
```
Comment (mic4ael) [inline indico/modules/events/editing/schemas.py:162]: This will silently error, since you are passing a `set` to the `dump` method which expects a single object.

File: indico/modules/events/editing/client/js/management/editable_type/EditableList.jsx (hunk 13 above)
Comment (ParthS007) [reply indico/modules/events/editing/client/js/management/editable_type/EditableList.jsx:46]: From the backend, data is coming in multiple json objects and I just destructure them to store in a list.

Comment (ThiefMaster) [reply indico/modules/events/editing/client/js/management/editable_type/EditableList.jsx:46]: But this line of code just shallow-copies a list...

Comment (mic4ael) [reply indico/modules/events/editing/client/js/management/editable_type/EditableList.jsx:46]: Yeah, you are just copying the array.  It is like a noop in this case

Comment (ParthS007) [reply indico/modules/events/editing/client/js/management/editable_type/EditableList.jsx:46]: oh okay, I got it. Thanks for the catch :+1:

File: indico/modules/events/editing/schemas.py (hunk 16 above)
Comment (ParthS007) [reply indico/modules/events/editing/schemas.py:162]: What should be the correct way to do it since editable will always be a single object?

Comment (ThiefMaster) [reply indico/modules/events/editing/schemas.py:162]: add a `get_editable(self, editable_type)` method to `Contribution` which contains e.g. this:

```python
return next((e for e in self.editables if e.type == editable_type), None)
```

File: indico/modules/events/editing/client/js/management/editable_type/EditableList.jsx (hunk 15 above)
Comment (ParthS007) [reply indico/modules/events/editing/client/js/management/editable_type/EditableList.jsx:56]: In `renderCell`, I am checking `sortedList[rowIndex][dataKey]` already, so I thought of adding info in the object itself.

Comment (mic4ael) [reply indico/modules/events/editing/client/js/management/editable_type/EditableList.jsx:56]: But it is a bit dirty. In case you e.g. access `contribution.code` later in the code, you would get `N/A` instead of null|undefined. IMO this shouldn't modify the data at all.

Comment (ThiefMaster) [reply indico/modules/events/editing/client/js/management/editable_type/EditableList.jsx:56]: Yes, please don't mutate data (unless it's something known to be regenerated each render).

Comment (ParthS007) [reply indico/modules/events/editing/client/js/management/editable_type/EditableList.jsx:54]: Considering this, should I remove the addition of `status` and `editor` attribute in object and add it in `renderCell` instead?

Comment (ThiefMaster) [reply indico/modules/events/editing/client/js/management/editable_type/EditableList.jsx:54]: yes, this is also mutating the object

File: indico/modules/events/editing/client/js/editing/timeline/StateIndicator.jsx (hunk 8 above)
[the remark this answers is not in the document]
Comment (panagiotappl) [reply indico/modules/events/editing/client/js/editing/timeline/StateIndicator.jsx:24]: I don't know. I can't think of anything better, so if no one has a different suggestion you can leave it as it is.

File: indico/modules/events/editing/blueprint.py [hunk 17]
```diff
@@ -45,6 +45,8 @@
                  common.RHEditingFileTypes)
 _bp.add_url_rule('/editing/api/<any(paper,slides,poster):type>/file-types', 'api_add_file_type',
                  management.RHCreateFileType, methods=('POST',))
+_bp.add_url_rule('/editing/api/<any(paper,slides,poster):type>/list', 'api_editable_list',
+                 editable_list.RHEditableList, methods=('GET',))
```
Comment (ThiefMaster) [inline indico/modules/events/editing/blueprint.py:49]: GET is the default, no need to specify `methods` at all

File: indico/modules/events/editing/client/js/management/editable_type/EditableList.jsx [hunk 18]
```diff
@@ -20,36 +21,65 @@ import {
...
+  });
+
+  const editableList = data || lastData;
+  if (isLoadingEditableList && !lastData) {
+    return <Loader inline="centered" active />;
+  } else if (!editableList) {
+    return null;
+  }
+  editableList.forEach(contribution => {
+    if (_.isEmpty(contribution.editable)) {
```
Comment (ThiefMaster) [inline indico/modules/events/editing/client/js/management/editable_type/EditableList.jsx:46]: `contribution.editable === null` should be enough here

File: indico/modules/events/editing/schemas.py [hunk 19]
```diff
@@ -139,6 +140,28 @@ class Meta:
...
+
+class EditingEditableListSchema(mm.ModelSchema):
+    class Meta:
+        model = Contribution
+        fields = ('id', 'friendly_id', 'title', 'code', 'editable')
+
+    editable = fields.Method('_get_editable')
+
+    def _get_editable(self, contribution):
+        editable_type = self.context["editable_type"]
```
Comment (ThiefMaster) [inline indico/modules/events/editing/schemas.py:160]: single quotes

File: indico/modules/events/editing/client/js/management/editable_type/EditableList.jsx (hunk 18 above)
Comment (ParthS007) [reply indico/modules/events/editing/client/js/management/editable_type/EditableList.jsx]: I have removed this, can you check again?

File: indico/modules/events/editing/client/js/management/editable_type/EditableList.jsx [hunk 20]
```diff
@@ -63,17 +80,28 @@ export default function EditableList() {
...
+          role="gridcell"
+          style={dataKey === 'title' ? {display: 'block'} : {}}
+        >
+          {dataKey === 'friendlyId' && row[dataKey]}
+          {dataKey === 'code' && code(row[dataKey])}
+          {dataKey === 'title' && row[dataKey]}
+          {dataKey === 'status' && (
+            <StateIndicator index={rowIndex} state={status(row['editable'])} circular label />
           )}
+          {dataKey === 'editor' && <Icon name="user outline" /> && row['editable'][dataKey]}
```
Comment (ThiefMaster) [inline indico/modules/events/editing/client/js/management/editable_type/EditableList.jsx:104]: What about a mapping object here?

```js
const renderFuncs = {
  code: renderCode,
  status: renderStatus,
  editor: renderEditor,
};

const data = sortedList[rowIndex][dataKey];
const fn = renderFuncs[dataKey] || (x => x);
```

And then here you can just do `{fn(data)}`

File: indico/modules/events/editing/controllers/backend/editable_list.py [hunk 21]
```diff
@@ -0,0 +1,28 @@
...
+
+
+class RHEditableList(RHEditableTypeManagementBase):
+    """Return the list of editables of the event for a given type"""
+    def _process_args(self):
+        RHEditableTypeManagementBase._process_args(self)
+        self.contributions = (Contribution.query
+                              .with_parent(self.event)
+                              .options(joinedload('editables'))
+                              .all())
```
Comment (ThiefMaster) [inline indico/modules/events/editing/controllers/backend/editable_list.py:24]: We should probably apply a default order here to have something consistent even if the user does not apply any order manually e.g. `.order_by(Contribution.friendly_id)`

File: indico/modules/events/editing/client/js/management/editable_type/EditableList.jsx [hunk 22]
```diff
@@ -63,17 +80,28 @@ export default function EditableList() {
     setSortedList(_sortList({sortBy, sortDirection}));
   };
 
+  const code = data => (!data ? Translate.string('N/A') : data);
+  const status = data => (_.isEmpty(data) ? 'not_submitted' : data.state);
```
[the remark this answers is not in the document]
Comment (ParthS007) [reply indico/modules/events/editing/client/js/management/editable_type/EditableList.jsx:84]: Yes, It's an empty object.

[the remark this answers is not in the document]
Comment (ThiefMaster) [reply indico/modules/events/editing/client/js/management/editable_type/EditableList.jsx:84]: That's strange. I'd expect `Nested` to give you `null` if the relationship is is `None`...

File: indico/modules/events/contributions/models/contributions.py [hunk 23]
```diff
@@ -564,6 +564,10 @@ def can_submit_proceedings(self, user):
         # Otherwise only users with submission rights are authorized
         return self.can_manage(user, 'submit', allow_admin=False, check_parent=False)
 
+    def get_editable(self, editable_type):
+        """Get the editable of given editable type."""
```
Comment (ThiefMaster) [inline indico/modules/events/contributions/models/contributions.py:568]: > Get the editable of the given type.

File: indico/modules/events/editing/client/js/management/editable_type/EditableList.jsx [hunk 24]
```diff
@@ -175,3 +206,9 @@ export default function EditableList() {
     </>
   );
 }
+
+EditableListDisplay.propTypes = {
+  editableList: PropTypes.array.isRequired,
+  editableType: PropTypes.string.isRequired,
+  eventId: PropTypes.any.isRequired,
```
Comment (ThiefMaster) [inline indico/modules/events/editing/client/js/management/editable_type/EditableList.jsx:213]: why `any`? this should always be a `number`..

File: indico/modules/events/editing/client/js/management/editable_type/EditableList.jsx [hunk 25]
```diff
@@ -175,3 +206,9 @@ export default function EditableList() {
     </>
   );
 }
+
+EditableListDisplay.propTypes = {
+  editableList: PropTypes.array.isRequired,
+  editableType: PropTypes.string.isRequired,
```
Comment (ThiefMaster) [inline indico/modules/events/editing/client/js/management/editable_type/EditableList.jsx:212]: `editableType: PropTypes.oneOf(Object.values(EditableType)).isRequired,`

File: indico/modules/events/editing/client/js/management/editable_type/EditableList.jsx [hunk 26]
```diff
@@ -175,3 +206,9 @@ export default function EditableList() {
     </>
   );
 }
+
+EditableListDisplay.propTypes = {
+  editableList: PropTypes.array.isRequired,
```
Comment (ThiefMaster) [inline indico/modules/events/editing/client/js/management/editable_type/EditableList.jsx:211]: `PropTypes.arrayOf(PropTypes.shape(...)).isRequired` (in the shape specify the props you are actually using).

File: indico/modules/events/editing/client/js/management/editable_type/EditableList.jsx (hunk 24 above)
Comment (ParthS007) [reply indico/modules/events/editing/client/js/management/editable_type/EditableList.jsx:213]: Ohh, I forgot to change this.

File: indico/modules/events/editing/client/js/management/editable_type/EditableList.jsx [hunk 27]
```diff
@@ -63,17 +81,42 @@ export default function EditableList() {
     setSortedList(_sortList({sortBy, sortDirection}));
   };
 
+  const renderCode = code => code || Translate.string('n/a');
+  const renderEditor = editable => {
+    !_.isEmpty(editable) ?? (<Icon name="user outline" /> && editable.editor.fullName);
```
Comment (ThiefMaster) [inline indico/modules/events/editing/client/js/management/editable_type/EditableList.jsx:86]: This line looks really weird. Why do we need null-coalescing here (does our babel config transpile this properly?), and the `&&` seems strange since the left side is never falsy...

Comment (ParthS007) [reply indico/modules/events/editing/client/js/management/editable_type/EditableList.jsx:86]: null-coalescing worked here but the the editor name was not visible because of `&&`. Changes that.

Comment (ThiefMaster) [reply indico/modules/events/editing/client/js/management/editable_type/EditableList.jsx:86]: but why would `_.isEmpty(editable)` ever return null?

File: indico/modules/events/editing/client/js/management/editable_type/EditableList.jsx [hunk 28]
```diff
@@ -20,36 +21,80 @@ import {
...
+  } else if (!editableList) {
+    return null;
+  }
+
+  let codePresent = false;
+  for (const contribution of editableList) {
+    if (contribution.code !== '') {
+      codePresent = true;
+    }
+  }
```
Comment (ThiefMaster) [inline indico/modules/events/editing/client/js/management/editable_type/EditableList.jsx:51]:
```js
const codePresent = Object.values(editableList).some(c => c.code);
```

File: indico/modules/events/editing/client/js/management/editable_type/EditableList.jsx [hunk 29]
```diff
@@ -63,17 +103,52 @@ export default function EditableList() {
...
       <TooltipIfTruncated>
-        <div index={rowIndex} styleName="rowcolumn-tooltip" role="gridcell">
-          {dataKey === 'assigned' && <Icon name="user outline" />}
-          {dataKey === 'status' ? (
-            <StateIndicator index={rowIndex} state={sortedList[rowIndex][dataKey]} circular label />
-          ) : (
-            sortedList[rowIndex][dataKey]
-          )}
+        <div
+          id={dataKey}
```
Comment (ThiefMaster) [inline indico/modules/events/editing/client/js/management/editable_type/EditableList.jsx:145]: is this ID needed for anything? it's adding tons of duplicate IDs..

Comment (ParthS007) [reply indico/modules/events/editing/client/js/management/editable_type/EditableList.jsx:145]: I don't think so. Removing it
